src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -272,6 +301,7 @@ public Map<String, RequestContentValidator.DataType> allowedKeys() {
                             .put("opendistro_security_roles", DataType.ARRAY)
                             .put("hash", DataType.STRING)
                             .put("password", DataType.STRING)
+                            .put("service", DataType.STRING)
```

Comment:
Maybe I missed this before, why is this needed? Service is part of the attributes field, not a top level field on the document

Reply:
I had to add because when doing the put request to create an account in the tests,  the application expected the 'service' entry to be on the root of the json,  and the 'service' entry inside 'attributes' wasn;t being used

![image](https://github.com/opensearch-project/security/assets/128482925/44844020-b30f-497f-b9b0-321ec7d69782)

Reply:
I'd recommend making a unit test to sent the expected response (with service in the attributes) and use that to troubleshot what isn't working.  @scrawfor99 Do you have any additional thoughts?

Reply:
https://github.com/opensearch-project/security/pull/2786/files#r1334459144